bin/test/main.ml

```diff
@@ -136,10 +135,9 @@ let err_eval ~cmd lines =
     lines;
   exit 1
 
-let eval_raw ?block ?root c ~line lines =
-  let test =
-    Toplevel.{ vpad = 0; hpad = 0; line; command = lines; output = [] }
-  in
+let eval_raw ?block ?root c lines =
+  let command = Location.mknoloc lines in
```

Comment:
I wonder if we could improve this so we get a loc at least for the prelude files. The prelude_str we can't do much about it unfortunately but I think we'll probably drop support for those in the future so I don't think it matters that much!

From a quick look `eval_raw` is only used in `eval_prelude` so I think we could make it so that `Mdx.Util.File.read_lines` could gather locations as well. If it's too widely used maybe we can extract an `Mdx.Util.File.read_lines_with_locations`. The idea is that we'd build proper locations for the prelude files and no loc for the prelude_str.

What do you think?

Reply:
That's a good idea, I modified the `read_lines` function to return a loc (the computation is a bit ad-hoc, but didn't find a way to do it another way) and added a few unit tests.